src/EditorFeatures/TestUtilities/CodeActions/CSharpCodeFixVerifier`2+Test.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions
+{
+    public static partial class CSharpCodeFixVerifier<TAnalyzer, TCodeFix>
+        where TAnalyzer : DiagnosticAnalyzer, new()
+        where TCodeFix : CodeFixProvider, new()
+    {
+        public class Test : CSharpCodeFixTest<TAnalyzer, TCodeFix, XUnitVerifier>
+        {
+            /// <summary>
+            /// By default, the compiler reports diagnostics for nullable reference types at
+            /// <see cref="DiagnosticSeverity.Warning"/>, and the analyzer test framework defaults to only validating
+            /// diagnostics at <see cref="DiagnosticSeverity.Error"/>. This map contains all compiler diagnostic IDs
+            /// related to nullability mapped to <see cref="ReportDiagnostic.Error"/>, which is then used to enable all
+            /// of these warnings for default validation during analyzer and code fix tests.
+            /// </summary>
+            private static readonly ImmutableDictionary<string, ReportDiagnostic> s_nullableWarnings = GetNullableWarningsFromCompiler();
+
+            public Test()
+            {
+                MarkupOptions = Testing.MarkupOptions.UseFirstDescriptor;
+
+                SolutionTransforms.Add((solution, projectId) =>
+                {
+                    var parseOptions = (CSharpParseOptions)solution.GetProject(projectId).ParseOptions;
+                    solution = solution.WithProjectParseOptions(projectId, parseOptions.WithLanguageVersion(LanguageVersion));
+
+                    var compilationOptions = solution.GetProject(projectId).CompilationOptions;
+                    compilationOptions = compilationOptions.WithSpecificDiagnosticOptions(compilationOptions.SpecificDiagnosticOptions.SetItems(s_nullableWarnings));
+                    solution = solution.WithProjectCompilationOptions(projectId, compilationOptions);
+
+                    var options = solution.Options;
+                    foreach (var (key, value) in Options)
+                    {
+                        options = options.WithChangedOption(key, value);
+                    }
+
+                    solution = solution.WithOptions(options);
```

Comment:
I believe this approach of options testing will not work for analyzers ported to CodeStyle layer. Solution/workspace options are currently not passed down to the `AnalyzerConfigOptionSet` which means they will be unavailable in CI. I understand we would like to eventually pass these down, but until that is implemented, these tests will not work in code style layer. This led to me eventually use the approach here: https://github.com/dotnet/roslyn/pull/41363/files#r378505952.

I can confirm this once this PR goes in, and I will try to rebase #41363 on top of this verifier. If required, I will port similar logic here.

Reply:
> I believe this approach of options testing will not work for analyzers ported to CodeStyle layer

I traced the original implementation through the base test classes and mirrored the approach (i.e. this is a direct translation of the original tests). Changes may be required for the CodeStyle layer, but even if necessary would be constrained to the two helpers.